File: Python/1679. Max Number of K-Sum Pairs.py
import collections


# noinspection PyMethodMayBeStatic
class Solution:
    def maxOperations(self, nums, k):
        nums = collections.Counter(nums)
        operations = 0

        for num, count in nums.items():
            if k - num in nums:
                operations += min(count, nums[k - num])

        return operations // 2

File: PHP/1679. Max Number of K-Sum Pairs.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $nums
     */
    public function maxOperations(array $nums, int $k): int
    {
        $nums = array_count_values($nums);
        $operations = 0;

        foreach ($nums as $num => $count) {
            if (array_key_exists($index = $k - $num, $nums)) {
                $operations += min($count, $nums[$index]);
            }
        }

        return intdiv($operations, 2);
    }
}
